cfgov/ask_cfpb/models/pages.py

```diff
@@ -241,6 +247,226 @@ def page_js(self):
+        """Return sorted nav items for sidebar."""
+        sorted_categories = [
+            {
+                'title': _(category.heading),
+                'url': "{}{}/".format(page.url, slug),
+                'active': (
+                    False if not page.portal_category
+                    else _(category.heading)
+                    == _(page.portal_category.heading))
+            }
+            for slug, category in self.category_map.items()
+        ]
+        return [{
+            'title': _(page.portal_topic.heading),
+            'url': page.url,
+            'active': False if page.portal_category else True,
+            'expanded': True,
+            'children': sorted_categories
+        }], True
+
+    @route(r'^$')
+    def portal_topic_page(self, request):
+        context = self.get_context(request)
+        heading = self.get_heading(self.portal_topic)
+        context['heading'] = heading
+        search_term = request.GET.get('search_term', '').strip()
+        if not search_term or len(unquote(search_term)) == 1:
+            count = self.query_base.count()
+            sqs = self.query_base
+            search_message = self.results_message(count, heading, search_term)
+        else:
+            sqs = self.query_base.filter(content=search_term)
+            count = sqs.count()
+            if count == 0:
+                # No results, so let's try to suggest a better query
+                suggestion = SearchQuerySet().models(
+                    AnswerPage).spelling_suggestion(search_term)
+                if suggestion == search_term:
+                    suggestion = None
+                elif (request.GET.get('correct', '1') == '1' and
+                        flag_enabled('ASK_SEARCH_TYPOS', request=request)):
+                    sqs = self.query_base.filter(content=suggestion)
+                    search_term, suggestion = suggestion, search_term
+                    count = sqs.count()
+            context['pages'] = sqs
+            search_message = self.results_message(count, heading, search_term)
+        paginator = Paginator(sqs, 10)
+        page_number = validate_page_number(request, paginator)
+        pages = paginator.page(page_number)
+        context.update({
+            'count': count,
+            'search_term': search_term,
+            'results_message': search_message,
+            'pages': pages,
+            'paginator': paginator,
+            'current_page': page_number,
+            'get_secondary_nav_items': self.get_nav_items,
+        })
+        return TemplateResponse(
+            request,
+            self.get_template(request),
+            context)
+
+    @route(r'^(?P<category>[^/]+)/$')
+    def portal_category_page(self, request, **kwargs):
+        context = self.get_context(request)
+        category_slug = kwargs.get('category')
+        self.portal_category = self.category_map.get(category_slug)
+        heading = self.get_heading(self.portal_category)
```

Comment:
I noticed that the portal category pages show the same title as the portal topic page. Could we either set the page's title using the heading value (or some variation on it) here so the [title tag will be correctly populated](https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/jinja2/v1/_layouts/base-common.html#L53) or override the template's `title` block to do the same thing?